vendor-extra/LiberoContentBundle/tests/ViewConverter/Inline/ItalicVisitorTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace tests\Libero\LiberoContentBundle\ViewConverter\Inline;
+
+use FluentDOM;
+use FluentDOM\DOM\Element;
+use Libero\LiberoContentBundle\ViewConverter\Inline\ItalicVisitor;
+use Libero\ViewsBundle\Views\View;
+use PHPUnit\Framework\TestCase;
+use tests\Libero\ContentPageBundle\ViewConvertingTestCase;
+
+final class ItalicVisitorTest extends TestCase
+{
+    use ViewConvertingTestCase;
+
+    /**
+     * @test
+     * @dataProvider nodeProvider
+     */
+    public function it_does_nothing_if_it_is_not_a_libero_italic_element(string $xml) : void
+    {
+        $visitor = new ItalicVisitor($this->createFailingInlineConverter());
+
+        $xml = FluentDOM::load("<foo>${xml}</foo>");
+        /** @var Element $element */
+        $element = $xml->documentElement->firstChild;
+
+        $newContext = [];
+        $view = $visitor->visit($element, new View(null), $newContext);
+
+        $this->assertNull($view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($newContext);
+    }
+
+    public function nodeProvider() : iterable
+    {
+        yield 'different namespace' => ['<i xmlns="http://example.com">foo</i>'];
+        yield 'different element' => ['<b xmlns="http://libero.pub">foo</b>'];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_is_not_the_italic_template() : void
+    {
+        $visitor = new ItalicVisitor($this->createFailingInlineConverter());
+
+        $xml = FluentDOM::load('<i xmlns="http://libero.pub">foo</i>');
+        /** @var Element $element */
+        $element = $xml->documentElement;
+
+        $newContext = [];
+        $view = $visitor->visit($element, new View('template'), $newContext);
+
+        $this->assertSame('template', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($newContext);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_text_set() : void
```

Comment:
I'm not sure how the `it_does_nothing_if_there_is_already_text_set` tests category works; it means that the content gets ignored? I suppose because there is an equivalent `text` instead, so should `'bar'` here be more similar to what it is overriding?

Reply:
Would making it `foo` be clearer? So you have a visitor with a high priority that has already processed the text (presumably different in some way) and haven't disabled this one (so the check stops it overwriting).

Reply:
Is the assumption that the text is only processed by one visitor too strong? You could have the title into a `<title>` and `<h1>` tags (silly example but I'm sure there are similar cases).

Reply:
There's two different things there. Processing the text for this case (yes, I'd expect only one visitor to do hit, hence these bailing out), and for the title appearing in multiple cases on the page (better example is the author list, which is in at least 3 places for eLife: content header, author information section and metadata).